File: 0007-reverse-integer/0007-reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let min= -Math.pow(2,31);
    let max=Math.pow(2,31)-1;

    let sign =x>0?1:-1

    let ans =parseInt(Math.abs(x).toString().split('').reverse().join(''))*sign

    if(ans<min || ans>max){
        return 0
    }
    else{
        return ans
    }
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
      min = (-2**31)
      max=(2**31)-1
      if x<0:
        sign=-1
      else:
        sign =1
      x=abs(x)
      res =int(str(x)[::-1])*sign
      
      if res<min or res>max:
        return 0
      else:
        return res
# In this case, both the time complexity (TC) and space complexity (SC) are O(log n)
